### DISCUSSION TITLE
Access to result of `action` inside of `unstable_shouldReload`

### DISCUSSION THREAD

**Discussion by kstoddard** [remix-run/remix#5009] [Proposals]
## Proposal

Expose result of `action` to `unstable_shouldReload`.

---

## Story

- Typical admin page with user list
- Each item in list has basic user details
- Each item also has an action to update the user details
- Create action also exists at top of list to add new users

**Scenario: JS disabled**

- Action to update user is clicked
- User is linked to edit page for user (e.g.., `/admin/users/1/edit`)
- Submission uses `action` at this route
...
- Action to create new user is clicked
- User is linked to create page (e.g., `/admin/users/create`)
- Submission uses `action` at this route

**Scenario: JS enabled**

- Action to update user is clicked
- Modal opens with edit form
- Form submits via fetcher at `/admin/users/1/edit`
...
- Action to create new user is clicked
- Modal opens with create form
- Form submits via fetcher at `/admin/users/create`

---

This workflow allows us to support Progressive Enhancement! However, in the "JS enabled" scenario, after a submission there is _always_ a re-sync (`loader` re-syncs data for user list). We'd like to instruct Remix NOT to perform this re-sync in instances when we already know the result of the `action` failed. In this case, we're saying "hey Remix, don't update this list, it's cool."

Is there already a way to do this with `unstable_shouldReload`? I see that there is access to the `url` inside of `unstable_shouldReload` so I guess it may be possible to add state to the URL after a failed action, and then perhaps decide based on that. Pretty sure that won't work since we'd have to get the result of the action in React land client side and then set the search param via useEffect (and I assume this is too late in the process).

Anyway, any ideas how to solve this problem?

**Comment by mikeybinns** [comment]
This is available since Remix 1.10.0 in the stable version of `unstable_shouldReload` which is now called `shouldRevalidate` 😃 
[Remix Release 1.10.0](https://github.com/remix-run/remix/releases/tag/remix%401.10.0)
[React Router shouldRevalidate docs](https://reactrouter.com/en/main/route/should-revalidate)

**Reply by kstoddard** [reply]
Hey @mikeybinns thanks for the reply! Did you mean to link out to React Router? This feature is still unstable in Remix and as far as I can tell you can’t access the result of an action inside of `unstable_shouldReload`.

https://remix.run/docs/en/v1/route/should-reload

**Reply by machour** [reply]
Someone needs to check out the 1.10.0 release note 🤭

**Reply by kstoddard** [reply]
@mikeybinns hey I overlooked the release notes (sorry) and yeah, just read them and you’re right! Looks like action data is now available. Awesome.

**Reply by mikeybinns** [reply]
As far as I can tell, it's stable since 1.10.0 in Remix. The Remix docs haven't been updated yet, which is why I linked out to React Router. As you can see in the release page I linked, Remix is now fully built on top of React Router and therefore the API is exactly the same. The release page also has the following line to back this up:

> If you have any questions on these new APIs, head on over to their official documentation in the [React Router docs](https://reactrouter.com/en/main).
